references.go

```diff
@@ -51,47 +47,73 @@ func (referencesTable) Schema() sql.Schema {
 	return RefsSchema
 }
 
-func (r *referencesTable) TransformUp(f sql.TransformNodeFunc) (sql.Node, error) {
-	return f(r)
+func (r *referencesTable) WithFilters(filters []sql.Expression) sql.Table {
+	nt := *r
+	nt.filters = filters
+	return &nt
 }
 
-func (r *referencesTable) TransformExpressionsUp(f sql.TransformExprFunc) (sql.Node, error) {
-	return r, nil
+func (r *referencesTable) WithProjection(colNames []string) sql.Table {
+	return r
```

Comment:
doesn't accept this table pushdowns of projections?

Reply:
we could maybe just remove this methods so the tables don't implement ProjctedTable

Reply:
@erizocosmico yep, that is a good idea.